feat(UI): Move filetransferwidget buttons side-by-side.

Chiitoo opened the pull request:
This aims to prevent accidental clicks on the incorrect button when accepting a file transfer, and the chatlog moves due to new messages being received at the time of the click.

Closes #2597.

Took only 14 months, but I finally got to it!  (Where does the time go...)

Originally I had only figured out how to move the buttons around, but I wanted to also make the left button rounded from the upper left corner, similar to what the 'Send file(s)' and 'Send a screenshot' buttons have going.  The way I managed to get it done feels a bit odd, as if drawing three different rectangles, but not quite?  Looking around it seems it's not that unusual.

The right button still uses the old code mostly, since it gets the lower right corner rounded for free due to the background.

![qtox_filetransferwidget_buttons_changed_02](https://cloud.githubusercontent.com/assets/9367616/22569766/719db9e6-e9a1-11e6-987e-0d1bc153d0cf.png)


(I wonder if this could be done by using css stylesheets instead, but I imagine there is a reason for it not having been done like so in the first place.)

I'm thinking with the extra space on top of the buttons, the file name and/or other information could be extended there, or something completely different, but that's for another pull request.

I will not be surprised at all if the code looks absolutely bonkers to someone who knows more about Qt and stuff than I do.  At least I finally got the idea out there, which was my main goal.  :]

Tested on Gentoo Linux and Windows.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/4122)
<!-- Reviewable:end -->

sudden6 commented:
Reviewed 1 of 3 files at r1.
Review status: 1 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/chatlog/content/filetransferwidget.h, line 72 at r1](https://reviewable.io:443/reviews/qtox/qtox/4122#-Kc0YwfFCBjx034BENDN:-Kc0YwfGq7bYiid8-0dp:b-r8y4wm) ([raw file](https://github.com/qtox/qtox/blob/7d578f24667b22661886eefedd38f7b1ad10eb96/src/chatlog/content/filetransferwidget.h#L72)):*
> ```objc
> 
> private slots:
>     void onLeftButtonClicked();
> ```

Can you maybe rename them according to their function and not according to their position?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4122)*
<!-- Sent from Reviewable.io -->

Chiitoo commented:
I was thinking about that, too, but couldn't make up my mind and left them closer to what they were.

Something like 'acceptButton' and 'cancelButton'?  Or longer, 'acceptTransferButton' and 'cancelTransferButton'?

With less buttons, 'acceptTransfer' and 'cancelTransfer'?

Something completely different?

I might like keeping the 'button' word there in some form, but that's me.

sudden6 commented:
`acceptButton` and `cancelButton` is fine for me

---

Review status: 1 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4122#-:-Kc3io0a_c24SO5Q42Xt:bxt3iy5)*
<!-- Sent from Reviewable.io -->

zetok commented:
on 7d578f2

buttons seem to be too small: 
![spectacle fn1992](https://cloud.githubusercontent.com/assets/3148759/22781348/03af0048-eeba-11e6-8aee-35fb0b8685fe.png)

Chiitoo commented:
You would prefer the buttons to reach the top from the bottom?  Or perhaps be the original size, but either centered, or on the bottom or on top?

It looked a bit... weird I think when I first tried them reaching the top from the bottom, and there is the idea of opening up some space for other things when they're smaller.

That said, I probably don't mind which way they will go.

zetok commented:
> Or perhaps be the original size, but either centered, or on the bottom or on top?

Original size sounds good, perhaps located on top?

ProMcTagonist commented:
Like this?

![new ft buttons mockup](https://cloud.githubusercontent.com/assets/4358079/22860926/5a2ebdc0-f0c0-11e6-961f-56ec7127b56a.png)

zetok commented:
@ProMcTagonist looks good.

Chiitoo commented:
That might be exactly the way I have it done already, but I want to look into some other changes while I'm at it (apologies for the delay, again).  One of them is letting the transfer progress go under the buttons, which might seem like a simple thing to do, but isn't quite so (or I'm not seeing the simple way yet, heh).

sudden6 commented:
@Chiitoo what do you think about merging this now to fix the urgent issue and doing the style improvements in another PR?

Chiitoo commented:
Yeah, I think it's about ready for now.

I put some time into figuring out ways to make the white lines around the buttons happen, as is in the original style (previously I just made the background go under the buttons).

The only thing missing visually, is making the progress bar go below the buttons, which seems to require a bit of re-making of the whole layout internally.

That can wait, but there is still also the naming-of-things thing I've been meaning to ask about.  I haven't been able to come up with a phrase that would catch all the functions the buttons have.  The only thing really in my mind, is leave them be 'leftButton' and 'rightButton' etc., and add a comment somewhere that makes it clear what they do...

Any ideas on that?

sudden6 commented:
If there's really no way we can name them according to their function, then name them 'leftButton' and 'rightButton'.

Chiitoo committed:
feat(UI): Move filetransferwidget buttons side-by-side.

This aims to prevent accidental clicks on the incorrect button in
cases where the chatlog moves at the time of the click.

Closes #2597.

Chiitoo commented:
Had a bit of merge conflict-fun with the recent changes, so it took even longer to get this ready.  Apologies for the delay yet again!.

The comment about the buttons and their functions is now after the includes.  Should it be above even those, or is this good?

sudden6 commented:
@Diadlo checks seem to fail on Travis for OSX with a mysterious error message, any idea?

Diadlo commented:
@sudden6 https://github.com/libcheck/check/issues/110 ?
Seems osx has old version of libcheck and [use clang](https://travis-ci.org/qTox/qTox/jobs/206224276#L65)
May be issue somehow related with adding clang-format?

sudden6 commented:
Maybe `clang-format` reformatted in a strange way? Or there have been some changes in Travis or the tools we use, because it worked a few days ago.

zetok commented:
@Chiitoo workaround has been merged, rebase the PR to make it pass travis.

sudden6 commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4122)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4122#-:-KeCqq7ctWybcISQt1y_:b-jbsf8z)*
<!-- Sent from Reviewable.io -->
